File: src/app/material-advisor/src/app/models/knowledge-check/topic/KnowledgeCheckQuestion.ts
import { LanguageText } from "@shared/models/LanguageText";
import { QuestionType } from "@shared/types/QuestionEnum";
import { KnowledgeCheckAnswerGroup } from "./KnowledgeCheckAnswerGroup";
import { GUID } from "@shared/types/GUID";

export interface KnowledgeCheckQuestion {
  id: GUID;
  number: number;
  type: QuestionType;
  content: LanguageText[];
  answerGroups: KnowledgeCheckAnswerGroup[];
}

File: src/api/MaterialAdvisor/MaterialAdvisor.Application/Models/KnowledgeChecks/Topics/KnowledgeCheckQuestion.cs
using MaterialAdvisor.Data.Enums;

namespace MaterialAdvisor.Application.Models.Topics;

public class KnowledgeCheckQuestion
{
    public Guid Id { get; set; }

    public byte Number { get; set; }

    public double Points { get; set; }

    public QuestionType Type { get; set; }

    public IEnumerable<LanguageText> Content { get; set; } = null!;

    public IEnumerable<KnowledgeCheckAnswerGroup> AnswerGroups { get; set; } = [];
}
